File: backend/flexikmservlet/src/main/java/alexpre/flexikm/servlet/service/SinistroService.java
package alexpre.flexikm.servlet.service;

import java.util.List;

import org.springframework.stereotype.Service;

import alexpre.flexikm.servlet.model.Sinistro;

@Service
public interface SinistroService
{
  public Sinistro insert(Sinistro sinistro);

  public List<Sinistro> getSinistri();

  public Sinistro update(Sinistro sinistro);

  public String deleteSinistroById(Long id);

  public List<Sinistro> getSinistroById(Long id);

  public List<Sinistro> getSinistriByIdPolizza(Long id);
}

File: frontend/src/app/services/sinistro.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Sinistro } from '../models/sinistro.model';

@Injectable({ providedIn: 'root' })
export class SinistroService {

  private baseUrl = '/Sinistro';

  constructor(private http: HttpClient) {}

  getAllByPolizza(polizzaId: number): Observable<Sinistro[]> {
    return this.http.get<Sinistro[]>(`${this.baseUrl}/getSinistriByIdPolizza/${polizzaId}`);
  }


  getById(id: number): Observable<Sinistro> {
    return this.http.get<Sinistro>(`${this.baseUrl}/getSinistroById/${id}`);
  }


  create(s: Sinistro): Observable<Sinistro> {
    return this.http.post<Sinistro>(`${this.baseUrl}/insSinistro`, s);
  }


  update(s: Sinistro): Observable<Sinistro> {
    return this.http.put<Sinistro>(`${this.baseUrl}/updSinistro`, s);
  }

  
  delete(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/deleteSinistroById/${id}`);
  }
}